### PR TITLE
feat(report): add support for SPDX

### PR DESCRIPTION
**Pull request by ShiraCohen33**
## Description
Add SPDX format support

## Related issues
- https://github.com/aquasecurity/trivy/issues/1819

## Checklist
- [X] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [X] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [X] I've added tests that prove my fix is effective or that my feature works.
- [X] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [X] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/report/spdx/spdx.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+	"github.com/spdx/tools-golang/spdx"
+	"github.com/spdx/tools-golang/tvsaver"
+	"k8s.io/utils/clock"
+
+	ftypes "github.com/aquasecurity/fanal/types"
+	"github.com/aquasecurity/trivy/pkg/types"
+)
+
+const (
+	SPDXVersion         = "SPDX-2.1"
```
Comment (seabass-labrax) [inline pkg/report/spdx/spdx.go:20]:
```suggestion
	SPDXVersion         = "SPDX-2.2"
```

The latest version of the [SPDX specification](https://spdx.github.io/spdx-spec/) is 2.2 :smiley:

**Review by seabass-labrax — CHANGES_REQUESTED** [review]
This looks really great; thanks for taking the time to implement this feature! I've left a small suggestion for this PR.

**Comment by seabass-labrax** [comment]
@ShiraCohen33 thanks for bumping the SPDX specification version! :smiley:

File: docs/docs/sbom/spdx.md [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+
+```
+$ trivy image --format spdx --output result.spdx alpine:3.15
+```
+
+<details>
+<summary>Result</summary>
+
+```
+$ cat result.spdx | jq .
```
Comment (knqyf263) [inline docs/docs/sbom/spdx.md:15]: Is this JSON?

Comment (ShiraCohen33) [reply docs/docs/sbom/spdx.md:15]: No, it's a mistake

**Review by knqyf263 — APPROVED** [review]

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**